extensions/parquet/table/src/main/java/io/deephaven/parquet/table/location/ParquetTableLocation.java

```diff
@@ -56,21 +54,24 @@ public class ParquetTableLocation extends AbstractTableLocation {
 
     private static final String IMPLEMENTATION_NAME = ParquetColumnLocation.class.getSimpleName();
 
+    private boolean isInitialized;
+    private volatile boolean isInitializedVolatile;
+
     private final ParquetInstructions readInstructions;
-    private final ParquetFileReader parquetFileReader;
-    private final int[] rowGroupIndices;
+    private ParquetFileReader parquetFileReader;
+    private int[] rowGroupIndices;
 
-    private final RowGroup[] rowGroups;
-    private final RegionedPageStore.Parameters regionParameters;
-    private final Map<String, String[]> parquetColumnNameToPath;
+    private RowGroup[] rowGroups;
+    private RegionedPageStore.Parameters regionParameters;
+    private Map<String, String[]> parquetColumnNameToPath;
 
-    private final TableInfo tableInfo;
-    private final Map<String, GroupingColumnInfo> groupingColumns;
-    private final List<DataIndexInfo> dataIndexes;
-    private final Map<String, ColumnTypeInfo> columnTypes;
-    private final List<SortColumn> sortingColumns;
+    private TableInfo tableInfo;
+    private Map<String, GroupingColumnInfo> groupingColumns;
+    private List<DataIndexInfo> dataIndexes;
+    private Map<String, ColumnTypeInfo> columnTypes;
+    private List<SortColumn> sortingColumns;
 
-    private final String version;
+    private String version;
```

Comment:
I'm tracing down this code more thoroughly... I want to make sure we group all the members that need initialization together. It's a bit obvious in this case, but probably worthwhile to be explicit with something like:

```java

    private volatile boolean isInitialized;
    
    // Access to all the following variables must be guarded by initialize()    
    private ParquetFileReader parquetFileReader;
    private int[] rowGroupIndices;
...
    private String version;    
    // -----------------------------------------------------------------
```